src/Chainweb/Pact/PactService.hs

```diff
@@ -902,14 +902,17 @@ readAccountGuard pdb account
 -- See: 'rewards/miner_rewards.csv'
 --
 minerReward
-    :: MinerRewards
+    :: ChainwebVersion
+    -> MinerRewards
     -> BlockHeight
     -> IO P.ParsedDecimal
-minerReward (MinerRewards rs q) bh =
-    case V.find (bh <=) q of
+minerReward v (MinerRewards rs q) bh = case V.find (bh <=) q of
```

Comment:
I suggest replacing this whole Vector+HashMap with just a `Map` --
- `V.find` is O(n) vs. `Map` `split` which is O(log n). 
- Less error prone (here we have to do two error checks)
- already sorted on key

Reply:
http://hackage.haskell.org/package/containers-0.6.0.1/docs/Data-Map-Strict.html#v:split

Reply:
As discussed in Slack, we made the `HashMap` a `Map.Strict`, and we're just making use of `lookupGE` now, which is O(log n)